MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+            }
+
+            #endregion
+
+            #region Int32 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int).
+            /// </summary>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next()
+            {
+                return (int)Remap(UInt64ToInt32(XORShift64Star()), int.MinValue, int.MaxValue, 0, int.MaxValue);
+            }
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next(int max)
+            {
+                return (int)Remap(UInt64ToInt32(XORShift64Star()), int.MinValue, int.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int32 (int) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next(int min, int max)
+            {
+                return (int)Remap(UInt64ToInt32(XORShift64Star()), int.MinValue, int.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Int64 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int64 that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int64</returns>
+            public long Next(long max)
+            {
+                return Remap(UInt64ToInt64(XORShift64Star()), long.MinValue, long.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int64 that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int64</returns>
+            public long Next(long min, long max)
+            {
+                return Remap(UInt64ToInt64(XORShift64Star()), long.MinValue, long.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #endregion
+
+            #region Private methods
+
+            private ulong XORShift64Star()
+            {
+                _lastValue ^= _lastValue >> 12;
+                _lastValue ^= _lastValue << 25;
+                _lastValue ^= _lastValue >> 27;
+                return unchecked(_lastValue * 2685821657736338717UL);
+            }
```

Comment:
Just use `_seed` here instead of `_lastValue`.  We don't need to keep the original seed value.  Think of `_seed` as the seed for the next random value to be generated.  If we wanted to know what the original seed value was, we would have generated it and given it to the constructor.